histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -0,0 +1,210 @@
+        'Haralick.SumAverage.Range',
+        'Haralick.SumVariance.Range',
+        'Haralick.SumEntropy.Range',
+        'Haralick.Entropy.Range',
+        'Haralick.Variance.Range',
+        'Haralick.DifferenceEntropy.Range',
+        'Haralick.IMC1.Range',
+        'Haralick.IMC2.Range',
+    ]
+
+    # compute object properties if not provided
+    if rprops is None:
+        rprops = regionprops(im_label)
+
+    # create pandas data frame containing the features for each object
+    numFeatures = len(feature_list)
+    numLabels = len(rprops)
+    fdata = pd.DataFrame(np.zeros((numLabels, numFeatures)),
+                         columns=feature_list)
+
+    # check for consistent shapes between 'I' and 'Label'
+    if im_intensity.shape != im_label.shape:
+        raise ValueError("Inputs 'I' and 'Label' must have same shape")
+
+    # determine if image is grayscale or RGB
+    if len(im_intensity.shape) != 2:  # color image
+        raise ValueError("Inputs 'I' should be a grayscale image")
+
+    for i in range(numLabels):
+        # get bounds of an intensity image
+        minr, minc, maxr, maxc = rprops[i].bbox
+        # grab nucleus mask
+        subImage = im_intensity[minr:maxr+1, minc:maxc+1].astype(np.uint8)
+        # gets GLCM or gray-tone spatial dependence matrix
+        arrayGLCM = graycomatrixext(subImage, offsets=offsets,
+                                    num_levels=num_levels,
+                                    gray_limits=gray_limits)
+        num_dims = len(subImage.shape)
+        # offsets
+        if offsets is None:
+            # set default offset value
+            offsets = np.identity(num_dims)
+        else:
+            # check sanity
+            if offsets.shape[1] != num_dims:
+                raise ValueError(
+                    'Dimension mismatch between input image and offsets'
+                )
+        num_offsets = offsets.shape[0]
+        f = np.zeros((numFeatures/2, num_offsets))
+        for r in range(num_offsets):
+            # normalize GLCM
+            R = np.sum(arrayGLCM[:, :, r], dtype=np.float)
+            nGLCM = arrayGLCM[:, :, r]/R
+            # get marginal-probability matrix summing the rows
+            px = np.sum(nGLCM, axis=1)
+            py = np.sum(nGLCM, axis=0)
+            # initialize marginal-probability matrix sets
+            # summing normalizedGLCM such that i+j = k or i-j = k
+            pxPlusy = np.zeros(2*num_levels-1)
+            pxMinusy = np.zeros(num_levels)
+            # arbitarily small positive constant to avoid log 0
+            e = 0.00001
+            for n in range(0, num_levels):
+                for m in range(0, num_levels):
+                    # gets marginal-probability matrix
+                    pxPlusy[n+m] = pxPlusy[n+m] + nGLCM[n, m]
+                    pxMinusy[abs(n-m)] = pxMinusy[abs(n-m)] + nGLCM[n, m]
+            # f0: computes angular second moment
+            f[0, r] = np.sum(np.square(nGLCM))
```

Comment:
@cdeepakroy " f " series is an array of the feature and offsets, which will be computed for mean and range of haralick features. Do you mean adding another panda dataframe to the ComputeHaralickFeatures.py?